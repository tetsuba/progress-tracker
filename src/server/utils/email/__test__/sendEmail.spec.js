const { getEmailMailOptions, sendMail } = require('../sendEmail')
jest.mock('nodemailer')
const nodemailer = require('nodemailer')

describe('@sendEmail', () => {
  describe('getEmailMailOptions', () => {
    it('should return the resetPassword html template and email subject', () => {
      const name = 'resetPassword'
      expect(getEmailMailOptions(name)).toMatchSnapshot()
    })

    it('should return the confirmEmail html template and email subject', () => {
      const name = 'confirmEmail'
      expect(getEmailMailOptions(name)).toMatchSnapshot()
    })

    it('should return a TypeError if a name is not provided', () => {
      expect.assertions(2)
      try {
        getEmailMailOptions()
      } catch (error) {
        expect(error).toBeInstanceOf(TypeError)
        expect(error).toHaveProperty(
          'message',
          "Cannot read property 'emailTemplateName' of undefined"
        )
      }
    })

    it('should return a TypeError if an incorrect name is not provided', () => {
      expect.assertions(2)
      try {
        getEmailMailOptions('incorrectName')
      } catch (error) {
        expect(error).toBeInstanceOf(TypeError)
        expect(error).toHaveProperty(
          'message',
          "Cannot read property 'emailTemplateName' of undefined"
        )
      }
    })
  })

  describe('sendMail', () => {
    it('should send an email', async () => {
      const sendMailMock = jest.fn()
      nodemailer.createTransport.mockReturnValue({ sendMail: sendMailMock })

      const token = '123456Token'
      const options = {
        html: `<html></html>`,
        subject: 'This is a subject',
      }

      await sendMail(token, options)
      expect(sendMailMock).toHaveBeenCalledTimes(1)
    })
  })
})
