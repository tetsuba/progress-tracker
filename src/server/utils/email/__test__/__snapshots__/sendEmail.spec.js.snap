// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`@sendEmail getEmailMailOptions should return the confirmEmail html template and email subject 1`] = `
Object {
  "html": "<html lang=\\"en\\">
<head>
    <meta http-equiv=\\"Content-Type\\" content=\\"text/html; charset=UTF-8\\">
    <meta name=\\"viewport\\" content=\\"width=device-width, initial-scale=1\\">
    <meta http-equiv=\\"X-UA-Compatible\\" content=\\"IE=edge\\">
    <title>Confirm Email Template</title>
    <style type=\\"text/css\\">
        body {
            font-family: Helvetica;
            background-color: #EFEFEF;
        }

        table.main {
            margin: 40px auto;
            border-color: #EEEEEE;
            background-color: #FFFFFF;
        }
    </style>
</head>
<body>
    <table cellpadding=\\"0\\" cellspacing=\\"0\\" border=\\"1\\" class=\\"main\\">
        <tr>
            <td>
                <table cellpadding=\\"40\\" cellspacing=\\"0\\" border=\\"0\\" align=\\"center\\">
                    <tr>
                        <td width=\\"500\\" valign=\\"top\\">
                            <h1>Progress Tracker</h1>                            <h3>Confirm your email address</h3>
                            <p>
                                Click on the link below to confirm your email address.
                            </p>
                            <a href=\\"confirm/{{token}}\\">Confirm my email address</a>
                        </td>
                    </tr>
                </table>
            </td>
        </tr>
    </table>
</body>
</html>",
  "subject": "Progress tracker confirm email address.",
}
`;

exports[`@sendEmail getEmailMailOptions should return the resetPassword html template and email subject 1`] = `
Object {
  "html": "<html lang=\\"en\\">
<head>
    <meta http-equiv=\\"Content-Type\\" content=\\"text/html; charset=UTF-8\\">
    <meta name=\\"viewport\\" content=\\"width=device-width, initial-scale=1\\">
    <meta http-equiv=\\"X-UA-Compatible\\" content=\\"IE=edge\\">
    <title>Confirm Email Template</title>
    <style type=\\"text/css\\">
        body {
            font-family: Helvetica;
            background-color: #EFEFEF;
        }

        table.main {
            margin: 40px auto;
            border-color: #EEEEEE;
            background-color: #FFFFFF;
        }
    </style>
</head>
<body>
    <table cellpadding=\\"0\\" cellspacing=\\"0\\" border=\\"1\\" class=\\"main\\">
        <tr>
            <td>
                <table cellpadding=\\"40\\" cellspacing=\\"0\\" border=\\"0\\" align=\\"center\\">
                    <tr>
                        <td width=\\"500\\" valign=\\"top\\">
                            <h1>Progress Tracker</h1>
                            <h3>Password reset request</h3>
                            <p>
                                Click on the link below to reset your password.
                            </p>
                            <a href=\\"reset/{{token}}\\">Reset my password.</a>
                        </td>
                    </tr>
                </table>
            </td>
        </tr>
    </table>
</body>
</html>",
  "subject": "Progress tracker password reset request.",
}
`;
